src/main/java/org/apache/commons/io/input/buffer/UnsyncBufferedInputStream.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.io.input.buffer;
+
+import java.io.IOException;
+import java.io.InputStream;
+import org.apache.commons.io.IOUtils;
+import static org.apache.commons.io.IOUtils.EOF;
+
+/**
+ * A BufferedReader class who does not care about thread safety, but very much faster.
+ *
+ * Should be able to replace java.io.BufferedReader in nearly every use-cases when you
+ * need the Reader be buffered, but do not need it have thread safety.
+ */
+public class UnsyncBufferedInputStream extends InputStream {
+    protected final InputStream inputStream;
+    protected final byte[] byteBuffer;
+
+    protected int nowIndex = 0;
+    protected int nowLimit = 0;
+
```

Comment:
Fields should be private, otherwise they become part of the API and cannot be changed easily.
They also make thorough testing much harder.

If necessary, getters and/or setters can be added.
But only if necessary, as they increase the testing requirements, and cannot be dropped once added.

Similarly elsewhere.

Reply:
@sebbASF
Hi.
for the protected fields:
    OK, I can change them to private.
for the public functions:
    they are already as less as can.

Reply:
AFAICT UnsyncBufferedInputStream does not need to expose any getter/setter methods, not does it need to expose eat().
As to UnsyncBufferedReader, AFAICT LineEndUnifiedBufferedReader could be rewritten to act as a Filter which would then make it more versatile.